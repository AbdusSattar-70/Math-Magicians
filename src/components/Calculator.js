import React, { useState } from 'react';

import '../index.css';
import Display from './Display';
import Keypad from './Keypad';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [displyValue, setDisplayValue] = useState(0);

  const handleCalculation = (event) => {
    setDisplayValue((displyValue) => calculate(displyValue, event.target.innerText));
  };

  return (
    <section className="container">
      <Display displyValue={displyValue.next || displyValue.total || '0'} />
      <Keypad eventListener={handleCalculation} />
    </section>
  );
};

export default Calculator;
