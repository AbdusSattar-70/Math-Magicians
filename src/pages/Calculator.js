import React, { useState } from 'react';

import Display from '../components/Display';
import Keypad from '../components/Keypad';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [displyValue, setDisplayValue] = useState(0);
  const handleCalculation = (event) => {
    setDisplayValue((displyValue) => calculate(displyValue, event.target.innerText));
  };

  return (
    <section className="container">
      <Display displyValue={displyValue.next || displyValue.total || '0'} />
      <Keypad eventListener={handleCalculation} />
    </section>
  );
};

export default Calculator;
