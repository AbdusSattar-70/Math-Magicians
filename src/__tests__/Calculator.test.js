import renderer from 'react-test-renderer';
import Calculator from '../pages/Calculator';

it('renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
