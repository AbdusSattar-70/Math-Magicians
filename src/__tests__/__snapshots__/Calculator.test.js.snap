// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="mainContainer"
>
  <h2>
    Let's Do Some Math! 
  </h2>
  <div
    className="container"
  >
    <div
      className="display"
    >
      0
    </div>
    <div
      className="keypad"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="deepOrange"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="deepOrange"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="deepOrange"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="deepOrange"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="deepOrange"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;
